src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,504 @@
+
+                    // if the identifier name conflicts with a different identifier that we've already seen
+                    if (allVarNames.some(ident => ident.identifier.text === node.text && ident.symbol !== symbol)) {
+                        const newName = getNewNameIfConflict(node, allVarNames);
+                        identsToRenameMap.set(symbolIdString, newName.identifier);
+                        synthNamesMap.set(symbolIdString, newName);
+                        allVarNames.push({ identifier: newName.identifier, symbol });
+                    }
+                    else {
+                        const identifier = getSynthesizedDeepClone(node);
+                        identsToRenameMap.set(symbolIdString, identifier);
+                        synthNamesMap.set(symbolIdString, { identifier, numberOfAssignmentsOriginal: allVarNames.filter(elem => elem.identifier.text === node.text).length, numberOfAssignmentsSynthesized: 0 });
+                        if ((isParameter(node.parent) && isCallbackOnTypePromise(node.parent.parent, setOfAllCallbacksToReturn)) || isVariableDeclaration(node.parent)) {
+                            allVarNames.push({ identifier, symbol });
+                        }
+                    }
+                }
+            }
+        });
+
+        function isCallbackOnTypePromise(child: Node, setOfAllCallbacksToReturn: Map<true>): boolean {
+            let node = child.parent;
+            if (isCallExpression(node) || isIdentifier(node) && !setOfAllCallbacksToReturn.get(getNodeId(node).toString())) {
+                const nodeType = checker.getTypeAtLocation(node);
+                const isPromise = nodeType && checker.getPromisedTypeOfPromise(nodeType);
+                return !!isPromise;
+            }
+
+            return false;
+        }
+
+        function deepCloneCallback(node: Node, clone: Node) {
+            const type = checker.getTypeAtLocation(node);
+            const symbol = checker.getSymbolAtLocation(node);
+            const renameInfo = symbol && identsToRenameMap.get(String(getSymbolId(symbol)));
+
+            if (renameInfo && originalType) {
+                const newIdentifier = identsToRenameMap.get(String(getSymbolId(symbol!)))!;
+                if (type) {
+                    originalType.set(newIdentifier.text, type);
+                }
+            }
+
+            if (setOfAllCallbacksToReturn) {
+                const val = setOfAllCallbacksToReturn.get(getNodeId(node!).toString());
+                if (val !== undefined) {
+                    setOfAllCallbacksToReturn.delete(getNodeId(node!).toString());
+                    setOfAllCallbacksToReturn.set(getNodeId(clone).toString(), val);
+                }
+            }
+        }
+
+        return getSynthesizedDeepClone(nodeToRename, /*includeTrivia*/ true, identsToRenameMap, /*setOfAllCallbacksToReturn,*/ checker, /*originalType*/ deepCloneCallback);

```

Comment:
Personally, I'd put this before the helper functions - it's easy to miss one line at the very end.